src/fides/api/service/connectors/saas_connector.py

```diff
@@ -621,80 +665,147 @@ def run_consent_request(
         )
         self.set_privacy_request_state(privacy_request, node, request_task)
         query_config = self.query_config(node)
+        saas_config = self.saas_config
+        fired: bool = (
+            False  # True if the SaaS connector was successfully called / completed
+        )
 
-        should_opt_in, filtered_preferences = should_opt_in_to_service(
-            self.configuration.system, privacy_request
+        notice_based_override_fn: Optional[RequestOverrideFunction] = (
+            self.obtain_notice_based_update_consent_fn_or_none(saas_config.type)
         )
 
-        if should_opt_in is None:
-            logger.info(
-                "Skipping consent requests on node {}: No actionable consent preferences to propagate",
-                node.address.value,
+        if notice_based_override_fn:
+            # follow the notice-based SaaS consent flow
+            notice_id_to_preference_map, filtered_preferences = (
+                build_user_consent_and_filtered_preferences_for_service(
+                    self.configuration.system,
+                    privacy_request,
+                    session,
+                    True,
+                )
             )
-            raise SkippingConsentPropagation(
-                f"Skipping consent propagation for node {node.address.value} - no actionable consent preferences to propagate"
+            if not notice_id_to_preference_map:
+                logger.info(
+                    "Skipping consent requests on node {}: No actionable consent preferences to propagate",
+                    node.address.value,
+                )
+                raise SkippingConsentPropagation(
+                    f"Skipping consent propagation for node {node.address.value} - no actionable consent preferences to propagate"
+                )
+            cache_initial_status_and_identities_for_consent_reporting(
+                db=session,
+                privacy_request=privacy_request,
+                connection_config=self.configuration,
+                relevant_preferences=filtered_preferences,
+                relevant_user_identities=identity_data,
+            )
+            fired = self._invoke_consent_request_override(
+                notice_based_override_fn,
+                saas_config.type,
+                self.create_client(),
+                policy,
+                privacy_request,
+                self.secrets,
+                identity_data,
+                notice_id_to_preference_map,  # type: ignore[arg-type]
+                self.build_notice_based_consentable_item_hierarchy(
```

Comment:
We should move this check earlier and avoid calling the `consent_update` altogether if the consentable items are `None`. I would treat this the same as not having any consent preferences.